velox/experimental/cudf/connectors/parquet/ParquetDataSource.cpp

```diff
@@ -144,18 +146,31 @@ std::optional<RowVectorPtr> ParquetDataSource::next(
     return nullptr;
   }
 
+  // Record start time before reading chunk
+  auto startTimeUs = getCurrentTimeMicro();
+  
+  std::unique_ptr<cudf::table> cudfTable;
   // Read a table chunk
   auto [table, metadata] = splitReader_->read_chunk();
-  auto cudfTable = std::move(table);
+  cudfTable = std::move(table);
   // Fill in the column names if reading the first chunk.
   if (columnNames_.empty()) {
     for (const auto& schema : metadata.schema_info) {
       columnNames_.emplace_back(schema.name);
     }
   }
-
+  
+ 
+  
+  auto* callbackData = new totalScanTimeCallbackData{startTimeUs, ioStats_};
```

Comment:
Is this being freed anywhere? Why not use an object instead?
`TotalScanTimeCallbackData  callbackData{startTimeUs, ioStats_};`